### ISSUE TITLE
add a check for undefined $fd in Gearman::Taskset

### ISSUE THREAD

**Issue by p-alik** [p-alik/perl-Gearman#9]
```bash
Use of uninitialized value $fd in vec at /path/to/build/Gearman-2.001.001-0/blib/lib/Gearman/Taskset.pm line 221.
Use of uninitialized value $fd in hash element at /path/to/build/Gearman-2.001.001-0/blib/lib/Gearman/Taskset.pm line 222.
Use of uninitialized value $fd in vec at /path/to/build/Gearman-2.001.001-0/blib/lib/Gearman/Taskset.pm line 221.
Use of uninitialized value $fd in hash element at /path/to/build/Gearman-2.001.001-0/blib/lib/Gearman/Taskset.pm line 222.
```

**Comment by p-alik** [comment]
> it occurs whenever the SSL connection from the client to the server is reset by the server.

**Comment by esabol** [comment]
I've been testing the IO::Select patch (at the above commits) to Taskset.pm with 2.004.001 for several days now, and I haven't encountered any problems.

**Comment by p-alik** [comment]
Thank you, @esabol. The changes are released in v2.004.003 to CPAN
